ParameterDataApi.yaml

```diff
@@ -88,122 +144,76 @@ components:
+          
+        data: 
+          description: A row of one or more data values, associated with a single discrete timestamp. Key/value pairs should be used, including units in the key name. The fields within this object should always follow a consistent order from row to row given a combination of messageType and messageVersion
+          type: object
+          additionalProperties: true    
+        
       required:
-        - timestamp
-        - fields
+        - timeOfApplicability
+        - data    
 
-    DataType:
-      type: string
-      enum:
-        - NUMBER
-        - INTEGER
-        - STRING
-        
     HeaderDefinition:
       type: object
       properties: 
-        startTime:
-          description: The start time for the time series
+      
+        componentUuid:
           type: string
-          format: date-time
-        
-        endTime: 
-          description: The end time for the time series
+          format: uuid
+          description: The unique identifier of the component within FRAIHMWORK that the parameter data associates with
+      
+        messageType:
+          type: string          
+          description: The type of the message, dictated by the client for delineating individual message formats
+
+        messageVersion:
           type: string
-          format: date-time
+          description: The version string for the parameter data messageType. If the client changes the data fields within a message stream, they should increment the version accordingly to distinguish message format along with the messageType
           
-        columns:
-          $ref: '#/components/schemas/ColumnDefinition'
-        
       required:
-        - startTime
-        - endTime
-        - columns
+        - componentUuid
+        - messageType
+        - messageVersion
+
+    SingleDataDefinition:
+      type: object
+      description: A definition of a set of parameter data having a single discrete timestamp. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys
+      allOf:
+      
+      - type: object
+        properties:          
+          header:
+            $ref: '#/components/schemas/HeaderDefinition'
+            
+      - $ref: '#/components/schemas/DataSample'
 
     TimeSeriesDataDefinition:
       type: object
-      description: A definition of time series data. This is loosely based upon Json Time Series, Version 1.0. (https://docs.eagle.io/en/latest/reference/index.html)
+      description: A time series collection for a set of parameter data. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys. It is encouraged that each element in the array has the same keys in the same order from data sample row to data sample row. Note: Time series data will be sorted by the server if not following order of occurence
```

Comment:
```suggestion
      description: A time series collection for a set of parameter data. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys. It is encouraged that each element in the array has the same keys in the same order from data sample row to data sample row. Note that time series data will be sorted by the server if not following order of occurrence
```

Reply:
The colon confuses the swagger editor, so it may lead to some unexpected generation results